include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4667,6 +4693,37 @@ struct OrtApi {
                   _In_reads_(num_external_initializer_files) char* const* external_initializer_file_buffer_array,
                   _In_reads_(num_external_initializer_files) const size_t* external_initializer_file_lengths,
                   size_t num_external_initializer_files);
+
+  /** \brief Append WebGPU execution provider to session options
+   *
+   * If WebGPU is not available, this function will return failure.
+   *
+   * \param[in] options
+   * \param[in] webgpu_options - specify the WebGPU provider options.
+   * \param[in] string_options_keys - keys to configure the string options
+   * \param[in] string_options_values - values to configure the string options
+   * \param[in] num_keys - number of keys passed in
+   *
+   * Supported keys are listed as below. All entries are optional.
+   *
+   * | Key                            | Possible Values                                | Default Value  |
+   * | ------------------------------ | ---------------------------------------------- | -------------- |
+   * | "preferredLayout"              | "NHWC" or "NCHW"                               | "NHWC"         |
+   * | "enableGraphCapture"           | "1" or "0"                                     | "0"            |
+   * | "storageBufferCacheMode"       | "disabled", "lazyRelease", "simple", "bucket"  | "bucket"       |
+   * | "uniformBufferCacheMode"       | "disabled", "lazyRelease", "simple", "bucket"  | "lazyRelease"  |
+   * | "queryResolveBufferCacheMode"  | "disabled", "lazyRelease", "simple", "bucket"  | "disabled"     |
+   * | "defaultBufferCacheMode"       | "disabled", "lazyRelease", "simple", "bucket"  | "disabled"     |
```

Comment:
enableGraphCapture is a general session option, not WebGPU specific.
To explain all these caches and different modes could be a little challenging. We should try our best to keep them internal.

Reply:
> enableGraphCapture is a general session option, not WebGPU specific.

I agree. But unfortunately, in current API this is EP specific configuration for:
- CUDA/TensorRT: `ProviderOptions["enable_cuda_graph"]`
- ROCM: `OrtROCMProviderOptions::enable_hip_graph`

considering backward compatibility this may not be easy change... @pranavsharma what do you think?

Reply:
no need to explain the details of each config entry - these should sit somewhere else in documentation.

however, keep a reference to a full list of available config entries is helpful.

Reply:
The graph capture functionality referred here is indeed specific to CUDA and TensorRT. Please do not make any backward incompat changes.